### ISSUE TITLE
Unable to resume training when using steps per epoch

### ISSUE THREAD

**Issue by akshit0201** [keras-team/keras#15538]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux
- TensorFlow installed from (source or binary):  binary
- TensorFlow version (use command below): 2.6.0
- Python version: 3.6.9
- Bazel version (if compiling from source):
- GPU model and memory: None

**Describe the problem**.
I'm using `steps_per_epoch` in my model.
After loading model and when retraining , the model loss is reset to inf.

**Describe the current behavior**.

I'm loading a pre-trained model , but tensorflow doesn't seem to remember it's previous model loss , so after the first epoch of retraining the model generated is taken as the  best model and written to memory. (I'm saving only the best model.)
This problem is occurring only when I use  `steps_per_epoch` in my model.

**Describe the expected behavior**.

The model should remember model loss from previous training.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): No
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**.

https://colab.research.google.com/drive/1Cu4I6iZAJ4pnOJ_Ks9i50e0qrwan4fEf?usp=sharing

**Source code / logs**.

When training this is the message displayed:
```Python
Epoch 00001: loss improved from inf to 0.00056, saving model to /gdrive/MyDrive/washing.ckpt
INFO:tensorflow:Assets written to: /gdrive/MyDrive/washing.ckpt/assets
```

When retraining this is the message displayed:
```Python
Epoch 00001: loss improved from inf to 0.00014, saving model to /gdrive/MyDrive/washing.ckpt
INFO:tensorflow:Assets written to: /gdrive/MyDrive/washing.ckpt/assets
```
